### PR TITLE
Add 'smoketests' targets.

### PR DESCRIPTION
**Pull request by eile**
See CommonSmokeTest.cmake for usage and documentation.

Also fixes project-install which was missing the installation of subprojects.

This will be used by CI.

### CODE REVIEW

File: Common.cmake [hunk 1]
```diff
@@ -128,6 +129,12 @@ if(NOT TARGET ${PROJECT_NAME}-all)
   set_target_properties(${PROJECT_NAME}-all PROPERTIES FOLDER ${PROJECT_NAME})
 endif()
 
+add_custom_target(${PROJECT_NAME}-install
+  ${CMAKE_COMMAND} -P ${PROJECT_BINARY_DIR}/cmake_install.cmake
+  DEPENDS ${PROJECT_NAME}-all)
+set_target_properties(${PROJECT_NAME}-install PROPERTIES
+  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/doxygen)
```
Comment (hernando) [inline Common.cmake]: What does the ON FOLDER do? I don't find the documentation that explains it.
Fix the C&P reference to doxygen in the path.

File: CommonSmokeTest.cmake [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#
+# Output targets
+# - smoketests: run all smoke tests of the given (sub)project
+# - <project>-smoketests: run all smoke tests of the given (sub)project
+# - <Target>-smoketest: run the given smoke test
+
+if(NOT TARGET smoketests)
+  add_custom_target(smoketests)
+  set_target_properties(smoketests PROPERTIES
+    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/tests)
```
Comment (hernando) [inline CommonSmokeTest.cmake]: Add the ON FOLDER property in CommonCTest.cmake also?

**Comment by eile** [comment]
Example use: https://github.com/Eyescale/hwsd/pull/61/commits/196849e1da8e1644704da2a0c5900ef976277175

**Comment by eile** [comment]
> we should somehow add this to CI

Read my comment above.

File: CommonSmokeTest.cmake (hunk 2 above)
Comment (eile) [inline CommonSmokeTest.cmake]: Not in this PR.

File: Common.cmake (hunk 1 above)
Comment (eile) [inline Common.cmake]: Read: 'EXCLUDE_FROM_DEFAULT_BUILD ON' && 'FOLDER ${PROJECT_NAME}/doxygen'

Will fix C&P

Comment (rdumusc) [inline Common.cmake]: Oh, I got confused by this in the past too ;-)
It should not be read as "ON FOLDER" but as: "EXCLUDE_FROM_DEFAULT_BUILD TRUE" "FOLDER ${PROJECT_NAME}/tests".
+1, FOLDER ${PROJECT_NAME}/doxygen does not look right (C&P error)

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # git master
 
+* [540](https://github.com/Eyescale/CMake/pull/540):
+  Add CommonSmokeTest.cmake
```
Comment (rdumusc) [inline CHANGES.md]: Elaborate a little, like:
Add CommonSmokeTest.cmake to check correct execution of installed applications

**Comment by rdumusc** [comment]
+1

**Pull request merged**